### DISCUSSION TITLE
Interactive Matplotlib plots in Notebooks

### DISCUSSION THREAD

**Discussion by bryanwweber** [executablebooks/meta#139] [Q&A] [answered]
Is there a way to retain the interactivity of `%matplotlib notebook` style plots? If I understand correctly (although there's a good chance I don't), they are implemented in pure Javascript in the output of a cell, so it should be possible, but when I use that mode and build my book, the figures are not interactive.

**Comment by akhmerov** [comment, marked as answer]
Unfortunately, `%matplotlib notebook`/`%matplotlib widget` uses communication with the kernel to regenerate the plot pngs, and therefore it won't work.

For interactive plots, you can try to use plotly/bokeh/vega or any other similar library.

**Comment by bryanwweber** [comment]
Thanks @akhmerov! I don't need interactivity with data, just the interactive display. In other words, I don't need to update the data in the plot, just change, say, the axis ranges or the zoom. It seems like Bokeh would be a good choice for that, specifically: https://docs.bokeh.org/en/latest/docs/user_guide/embed.html#components

**Reply by akhmerov** [reply]
I realize that. Still, matplotlib only has a python rendering engine. I've used both plotly and bokeh for rendering of static content, both were quite OK.

**Reply by bryanwweber** [reply]
:+1: Thanks! There's also this doc for Jupyter + Bokeh: https://docs.bokeh.org/en/latest/docs/user_guide/jupyter.html
